CONTRIBUTING.md

```diff
@@ -91,6 +93,23 @@ This should also pass for all pre-release features:
 cargo clippy --all-features
 ```
 
+#### Extensions
+
+First-party extensions are written in TypeScript. Extensions code must be
+formatted with [prettier](https://prettier.io/). Install it via `npm`:
+
+```sh
+npm install -g prettier
+```
```

Comment:
Installation isn't documented for the Rust side either and I don't think it's necessary. Some people might use `npm`, others use a different mechanism. All possible options should be way better documented by prettier so we don't need to duplicate it here. That way things are also more future-proof.

Reply:
The [install docs](https://prettier.io/docs/en/install.html) only cover the use case of a pure JS codebase and a non-global install, which is not our case. We could either remove the install part and let the users figure it out for themselves, or just nudge them via rewording (i.e. _here's a way you could possibly get this done_) but linking to the original documentation is not only not future proof but also confusing _right now_.

Reply:
I would just let the user figure it out. It really shouldn't be that difficult. I'd install it from the official Arch repos for example, so the docs wouldn't even apply to all our current uses.

It's just that excessive verbosity in documentation is evil and I don't see much benefit to the statement right now, since nobody should be stumped by the task of installing prettier.

Reply:
>avoid any unnecessary details that are documented in other places anyway (like installing prettier for example

As I said above, due to how our repo is structured, prettier _doesn't_ provide that specific piece of documentation and thus a user would be left wondering how to fit prettier's recommended installation workflow (i.e. add it locally as a dev-dependency) into our repo's structure.

>Especially considering that a software developer who can't figure out how to install prettier probably won't be able to send a patch that requires doing so anyway.

That's a flawed assumption, though one that I would have agreed with in the past. Most users who are faced with a list of instructions with a missing step are just going to either bail unless they are motivated to see something through, which is a disservice to many potential contributors, or open an issue to ask for clarifications, which is a waste of everybody's time.

This doesn't automatically mean that those users are not smart enough, though, just that they are lacking proper context.

I agree that verbosity is mostly undesirable, but having a _tiny bit_ more information often goes a long way into ensuring some legitimate doubts, however small, never even crop up.

Reply:
Anyway, I reworded the docs ever so slightly to make the concept more "optional" without implying that `npm` is the only intended way.